test/e2e/utils.go

```diff
@@ -316,7 +317,15 @@ func waitForDNSConditions(t *testing.T, cl client.Client, timeout time.Duration,
 		}
 		expected := operatorConditionMap(conditions...)
 		current := operatorConditionMap(dns.Status.Conditions...)
-		return conditionsMatchExpected(expected, current), nil
+
+		if conditionsMatchExpected(expected, current) {
```

Comment:
What the reason that we all of the sudden need this retry logic? What about a good  #comment to describe your motivation for future developers?